feat(upgrade): deprecated minFreeDiskSpaceGB and add skipGarbageCollectionThresholdCheck (#706)
harvester/docs: docs/upgrade/automatic.md

## Free system partition space requirement

_Available as of v1.5.0_

> During upgrade, harvester loads images on each node. If the disk usage is more than garbage collection threshold in kubelet, kubelet will delete unused images to free up space. This may break airgapped environment, because the images are not available on the node.

Reviewer: ```suggestion
Harvester loads images on each node during upgrades. When disk usage exceeds the kubelet's garbage collection threshold, the kubelet deletes unused images to free up space. This may cause issues in airgapped environments because the images are not available on the node.
```

> In harvester v1.5.0, we add checks to ensure that each node doesn't trigger garbage collection after loading new images.

Reviewer: ```suggestion
Harvester v1.5.0 includes checks that ensure nodes do not trigger garbage collection after loading new images.
```

![](/img/v1.5/upgrade/upgrade_free_space_check.png)

> If some nodes do not have enough free system partition space, but you still want to try upgrading, you can customize the upgrade by updating the `harvesterhci.io/skipGarbageCollectionThresholdCheck: true` annotation of `Version` object.

Reviewer: ```suggestion
If you want to try upgrading even if the free system partition space is insufficient on some nodes, you can update the `harvesterhci.io/skipGarbageCollectionThresholdCheck: true` annotation of the `Version` object.
```

![](/img/v1.2/upgrade/upgrade_free_space_check.png)
![](/img/v1.5/upgrade/upgrade_free_space_check.png)

If some nodes do not have enough free system partition space, but you still want to try upgrading, you can customize the upgrade by updating the `harvesterhci.io/minFreeDiskSpaceGB` annotation of `Version` object.
If some nodes do not have enough free system partition space, but you still want to try upgrading, you can customize the upgrade by updating the `harvesterhci.io/skipGarbageCollectionThresholdCheck: true` annotation of `Version` object.

```
apiVersion: harvesterhci.io/v1beta1
kind: Version
metadata:
  annotations:
    harvesterhci.io/minFreeDiskSpaceGB: "30" # the value is pre-defined and may be customized

>   name: 1.2.0

Reviewer 2: Do we need to update the doc in v1.3 (v1.3.2) and v1.4 (v1.4.0? not sure about that) to point out that `harvesterhci.io/minFreeDiskSpaceGB: "30"` is not available anymore?

Author: The issue is targeted on harvester v1.5.0, so it will be used from v1.5.0 to v1.5.1, not v1.4.x to v1.5.0.

Reviewer 2: I mean `harvesterhci.io/minFreeDiskSpaceGB` was broken after this change https://github.com/harvester/harvester/commit/9cd8124bbfc1abf0e01e5a5d9de71014a25cd935. But since there is already a GH issue: https://github.com/harvester/harvester/issues/7131, perhaps no need to update the v1.3, v1.4 doc.
